### PR TITLE
BUG-108475 Flex subscription organization related changes

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/FlexSubscription.java [hunk 1]
```diff
@@ -10,11 +10,13 @@
 import javax.persistence.Table;
 import javax.persistence.UniqueConstraint;
 
+import com.sequenceiq.cloudbreak.authorization.OrganizationResource;
 import com.sequenceiq.cloudbreak.domain.organization.Organization;
+import com.sequenceiq.cloudbreak.domain.organization.OrganizationAwareResource;
 
 @Entity
 @Table(uniqueConstraints = @UniqueConstraint(columnNames = {"account", "name"}))
```
Comment (mhmxs) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/FlexSubscription.java]: Uniq contraint not need any more

File: core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java [hunk 2]
```diff
@@ -34,62 +37,92 @@
...
+    protected OrganizationResource resource() {
+        return OrganizationResource.FLEXSUBSCRIPTION;
+    }
+
+    @Override
+    protected boolean canDelete(FlexSubscription resource) {
+        if (resource == null) {
+            throw new NotFoundException("FlexSubscription could not be found.");
+        }
+        if (stackService.countByFlexSubscription(resource) != 0L) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java]: else if would be better

File: core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java [hunk 3]
```diff
@@ -34,62 +36,88 @@
...
-    public FlexSubscription findByNameInAccount(String name, String owner, String account) {
-        LOGGER.info("Looking for Flex subscription with name: {}, in account: {}", name, account);
-        return flexSubscriptionRepository.findPublicInAccountByNameForUser(name, owner, account);
+    public Set<FlexSubscription> findAllForUser(IdentityUser user) {
+        Organization organization = getOrganizationService().getDefaultOrganizationForUser(user);
+        return findAllForUserAndOrganization(user, organization.getId());
     }
 
-    public List<FlexSubscription> findPublicInAccountForUser(IdentityUser user) {
+    public Set<FlexSubscription> findAllForUserAndOrganization(IdentityUser user, Long organizationId) {
```
Comment (attilapalfi92) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java]: this identityuser should be removed from here.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java [hunk 4]
```diff
@@ -125,7 +153,8 @@ private void updateSubscriptionsDefaultFlagsIfNeeded(FlexSubscription subscripti
     }
 
     private void setFlexSubscriptionFlag(String name, IdentityUser identityUser, BiConsumer<FlexSubscription, Boolean> setter) {
-        List<FlexSubscription> allInAccount = flexSubscriptionRepository.findAllByAccount(identityUser.getAccount());
+        Organization organization = getOrganizationService().getDefaultOrganizationForUser(identityUser);
```
Comment (attilapalfi92) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/flex/FlexSubscriptionService.java]: identityUser is uncessary

**Pull request merged**